thunder/core/vjp_utils.py

```diff
@@ -47,15 +47,14 @@ def make_aug_forward_and_backward(bsym: BoundSymbol) -> tuple[Callable, Callable
     """
     import thunder
     from thunder.common import _make_cache_key
-    from thunder.core.transforms import _get_gradfn, eval_trace
+    from thunder.core.transforms import _get_gradfn_and_executor, eval_trace
 
-    joint_forward_backward = _get_gradfn(bsym)
+    joint_forward_backward, executor = _get_gradfn_and_executor(bsym)
     utils.check(
         joint_forward_backward is not None,
         lambda: f"Cannot generate forward and backward functions for {bsym.sym.name}",
     )
-
-    key = (bsym.sym, subkey := _make_cache_key(bsym.args, bsym.kwargs))
+    key = (bsym.sym, subkey := _make_cache_key(bsym.args, bsym.kwargs), executor)
```

Comment:
Here's a small readability improvement and potential optimization in the case of a cache miss. I don't know the details but I think to determine a cache miss faster it's better to put elements that have a higher chance to change as the first ones.
```suggestion
    key = (bsym.sym, executor, subkey := _make_cache_key(bsym.args, bsym.kwargs))
```